Readme.txt

===================
ROBBIE STRIKES BACK
===================
Amstrad CPC 464/6128

Some decades later, Robbie the robot, already retired, has come back to his garden and passes his time growing his favorite flower, the “Thyrgodian Megga Chrisanthodil”.
But again the garden is invaded by evil and opportunist insects.
Now, as before, Robbie is ready to act!

Robbie has less oil in his gears nowadays, but the same decision.
He has got three lethal insecticides to fight all living bugs.

So, he must sprinkle each insect that comes near his flowers with the right spray.
A wrong spray will get insect momentarily paralised, or they won’t suffer at all.
If Robbie leaves his flowers neglected, insect will eat them in seconds!

Note that in level five, flowering plants will reveal Robbie’s dreams!
Insects’ speed will increase each five levels.


NEW ENEMIES:

Greasy Worms: They move slowly in a horizontal way.
They are the first to appear. [35 points]

Hairy Aphids: They move very fast in a diagonal way,
sweeping away all the garden in seconds. [50 points]

Sucker Wasps: The most dangerous of all, because they will attack Robbie directly! [75 points]


ITEMS:

Fly Swatter: When you use it, it kills all insects in the garden. [100 points]

Watering Can: When you use it, it speeds up the plants’ growing. [100 points]

Fertilizer: When you use it, it speeds up the plants’ growing twice as much the watering can. [100 points]


GAME CONTROLS:

Setting 1		Setting 2		Setting 3
----------------        ----------------        ----------------
Q = up			W = up			Joystick
A = down		S = down		M = music ON/OFF
O = left		A = left		X = finish
P = right		D = right
SPC = fire		SPC = fire
M = music ON/OFF	M = music ON/OFF
X = finish		X = finish


LOADING INSTRUCTIONS (CPC464):

1) Reset the computer pressing down CTRL + SHIFT + SPC simultaneously and following this order.

2) Insert the tape and rewind it from the beginning.

3) Press simultaneously CTRL + ENTER (little) keys. 
You will see “PRESS PLAY THEN ANY KEY” on the screen. 
Press PLAY on the tape recorder and then press any key. 
The game will get 5 minutes to load completely.


 ROBBIE STRIKES BACK! 
 Copyleft (C) 2017

 SalvaKantero:	Program, graphics, FX sound
       Beyker:	Music
  Masterklown:	Tape cover
 Felipe Monge:	Additional graphics
 Dany Cantero:	Translations



History.txt
ROBBIE STRIKES BACK
===================

v1.2 (10/05/18) REV:3
=====================
01/05/18 - Se limpia la pantalla de sprites al florecer la planta, para que no queden parcialmente borrados
02/05/18 - Se recortan las sombras de los sprites para que se muestren mejor en monitores de fósforo verde
02/05/18 - Aumenta la velocidad de gusanos y pulgones
05/05/18 - Mejora la detección de colisiones
08/05/18 - Los enemigos resucitan a una distancia segura de Robbie
10/05/18 - Mejora la sincronización del segundo disparo con respecto al primero

v1.1 (02/01/18) REV:2
=====================
02/01/18 - Se repara una incidencia con los marcadores al sobrepasar 9999 puntos

v1.0 (01/01/18)
===============
31/12/17 - Pantalla de carga para el CDT
22/12/17 - Carátula
05/11/17 - Música SI/NO
30/10/17 - Robbie no se para mientras dispara
28/09/17 - Música en menú
28/09/17 - Música al florecer la planta
24/09/17 - Mantener la posición de sprays al morir Robbie 
18/09/17 - Sonidos FX
19/08/17 - Timing
02/08/17 - Ciclo del juego (inicio-niveles-game over)
11/07/17 - Optimizar ancho de la zona de juego
10/07/17 - Mejorar aleatoriedad
08/07/17 - Gestión de crecimiento de la planta
25/06/17 - Gráficos de la planta
18/06/17 - Gestión de objetos
11/06/17 - Optimización de la velocidad (se evitan las máscaras, fondo liso)
22/05/17 - Gráficos de objetos
21/05/17 - Disparos de Robbie
14/05/17 - Gestión de vidas restantes, puntos y gameover
14/05/17 - Sistema de colisiones con los sprites enemigos
13/05/17 - Generador de enemigos en pantalla
13/05/17 - 4 instancias de enemigos por pantalla que pueden ser del mismo tipo.
13/05/17 - Movimiento del sprite gusano
12/05/17 - Movimiento del sprite pulgón
12/05/17 - Movimiento del sprite avispón
09/05/17 - Gráficos de sprites enemigos
07/05/17 - Paredes detienen a los sprites
05/05/17 - Diseño definitivo de la zona de marcadores
05/05/17 - Menú inicial
03/05/17 - Caracteres gráficos alfanuméricos personalizados
28/04/17 - Movimientos de Robbie
28/04/17 - Configurar teclado y Joystick


cdt/screen.c
//-----------------------------LICENSE NOTICE------------------------------------
//  This file is part of CPCtelera: An Amstrad CPC Game Engine
//  Copyright (C) 2015 ronaldo / Fremos / Cheesetea / ByteRealms (@FranGallegoBR)
//
//  This program is free software: you can redistribute it and/or modify
//  it under the terms of the GNU Lesser General Public License as published by
//  the Free Software Foundation, either version 3 of the License, or
//  (at your option) any later version.
//
//  This program is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//  GNU Lesser General Public License for more details.
//
//  You should have received a copy of the GNU Lesser General Public License
//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
//------------------------------------------------------------------------------

#include <cpctelera.h>
#include "RSB1.h"
#include "RSB2.h"

//PALETTE={18 14 12 15 10 16 6 13 24 9 19 25 20 3 2 7}

const u8 PAL[16] =
{
    0x12, 0x0E, 0x0C, 0x0F, 0x0A, 0x10, 0x06, 0x0D,
    0x18, 0x09, 0x13, 0x19, 0x14, 0x03, 0x02, 0x07
};

void main(void) 
{
    cpct_disableFirmware();
    cpct_setVideoMode(0);
    cpct_fw2hw(PAL, 16);
    cpct_setPalette(PAL, 16);
    cpct_setBorder(PAL[9]);
    cpct_clearScreen(cpct_px2byteM0(1,1));
    cpct_drawSprite(g_RSB1_0, CPCT_VMEM_START, 20, 200);
    cpct_drawSprite(g_RSB1_1, CPCT_VMEM_START + 20, 20, 200);
    cpct_drawSprite(g_RSB2_0, CPCT_VMEM_START + 40, 20, 200);
    cpct_drawSprite(g_RSB2_1, CPCT_VMEM_START + 60, 20, 200);
    while (1);
}
